script.py

```diff
@@ -0,0 +1,133 @@
+        for root, _, files in os.walk(outpath):
+            if (root == "admin") or (root == "config") or (root == "local"):
+                continue
+            for file in files:
+                local_path = os.path.join(root, file)
+                if file.endswith(".metadata.json"):
+                    key = (
+                        f"{path}/{os.path.split(root)[-1]}/{file.split('.')[0]}/{file}"
+                    )
+                else:
+                    key = f"{path}/{os.path.split(root)[-1]}/{os.path.splitext(file)[0]}/{file}"
+                s3.upload_file(local_path, bucket, key)
+                logging.info(f"File uploaded successfully: {key}")
+    except Exception as e:
+        logging.error(f"An error occurred: {e}")
+
+
+def download_from_s3(bucket, folder, s3objpath):
+    s3 = boto3.client("s3")
+    try:
+        objects = s3.list_objects_v2(Bucket=bucket, Prefix=folder)["Contents"]
+        for obj in objects:
+            key = obj["Key"]
+            if not os.path.exists(
+                (os.path.join(s3objpath, key.split("/")[0], key.split("/")[1]))
+            ):
+                os.makedirs(
+                    os.path.join(s3objpath, key.split("/")[0], key.split("/")[1])
+                )
+            with open(
+                os.path.join(
+                    s3objpath, key.split("/")[0], key.split("/")[1], key.split("/")[3]
+                ),
+                "wb",
+            ) as f:
+                s3.download_fileobj(bucket, key, f)
+            logging.info("file with key: " + key + " downloaded from s3")
+    except Exception as e:
+        logging.error(f"Backup failed: {e}")
+
+
+def mongorestore(uri, rfolder):
+    try:
+        command = ["mongorestore", "--uri", uri, rfolder]
+        subprocess.run(command, check=True)
+        logging.info("Restore completed successfully.")
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Restore failed: {e}")
+
+
+def cleanup_backups(bucket):
+    s3 = boto3.client("s3")
+    four_days_ago = datetime.now() - timedelta(days=4)
+    four_days_ago = four_days_ago.date()
+    try:
+        for i in range(1, 4):
+            folder = s3.list_objects_v2(Bucket=bucket, Prefix=f"{four_days_ago}_{i}")
+            if "Contents" in folder:
+                boto3.resource("s3").Bucket(bucket).objects.filter(
+                    Prefix=f"{four_days_ago}_{i}"
+                ).delete()
+            else:
+                logging.info(f"No backup exist for date {four_days_ago}_{i}.")
+    except Exception as e:
+        logging.error(f"Backup cleaning failed: {e}")
+
+
+if __name__ == "__main__":
+
+    source_uri = os.getenv("source_uri")
+    outpath = os.getenv("outpath")
+    bucket = os.getenv("bucket")
+    folder = os.getenv("folder")
+    s3objpath = os.getenv("s3objpath")
+    destination_uri = os.getenv("destination_uri")
+    max_backups = os.getenv("max_backups")
+    rfolder = os.getenv("restorefolder")
```

Comment:
this is also static
i want to run the script automatically everyday
so think of it from the server's perspective
no one will change the env vars in server everyday
kindly make this value dynamic